coderd/templates.go

```diff
@@ -616,6 +616,76 @@ func (api *API) templateDAUs(rw http.ResponseWriter, r *http.Request) {
 	httpapi.Write(ctx, rw, http.StatusOK, resp)
 }
 
+// @Summary Get template app usage by ID
+// @ID get-template-app-usage-by-id
+// @Security CoderSessionToken
+// @Produce json
+// @Tags Templates
+// @Param template path string true "Template ID" format(uuid)
+// @Param from query string false "Filter by app usage starting from this date"
+// @Param to query string false "Filter by app usage until this date"
+// @Success 200 {object} codersdk.TemplateAppUsageResponse
+// @Router /templates/{template}/app-usage [get]
+func (api *API) appUsage(rw http.ResponseWriter, r *http.Request) {
+	var err error
+	ctx := r.Context()
+	template := httpmw.TemplateParam(r)
+	if !api.Authorize(r, rbac.ActionRead, template) {
+		httpapi.ResourceNotFound(rw)
+		return
+	}
+
+	// Default date filter values
+	from := time.Now().Add(-24 * 7 * time.Hour) // Default is last 7 days
```

Comment:
nit: Not sure about this, but we might be using `since`/`to` in the codebase. It might be good idea to rename this.